pkg/dns/azure/client/auth.go

```diff
@@ -1,27 +1,27 @@
 package client
 
 import (
-	"errors"
+	"fmt"
 	"os"
+	"path/filepath"
 	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/fsnotify/fsnotify"
 	"github.com/jongio/azidext/go/azidext"
+	"k8s.io/klog/v2"
 )
 
 func getAuthorizerForResource(config Config) (autorest.Authorizer, error) {
 	var cloudConfig cloud.Configuration
 	switch config.Environment {
 	case azure.ChinaCloud:
 		cloudConfig = cloud.AzureChina
-	// GermanCloud was closed on Oct 29, 2021
-	// https://learn.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud
-	// case azure.GermanCloud:
-	// return nil, nil
```

Comment:
@Miciah Which other reason do you need other than `"// GermanCloud was closed on Oct 29, 2021"`. The new `azidentity` doesn't even have a config for `GermanCloud` as it does for the other clouds.

Reply:
Yeah I had just removed it since there is no German cloud and I hadn't seen a message like this in the other repos in similar code. Basically removed it as clean up. Happy to put it back if you want.

Reply:
Thanks!  Yeah, let's keep the comment then.